container.go

```diff
@@ -56,40 +56,52 @@ func containerFromRecord(client *Client, c containers.Container) *container {
 var _ = (Container)(&container{})
 
 type container struct {
-	mu sync.Mutex
+	mu sync.RWMutex
+	id string
 
 	client *Client
 	c      containers.Container
 }
 
 // ID returns the container's unique id
 func (c *container) ID() string {
+	c.mu.RLock()
+	defer c.mu.RUnlock()
 	return c.c.ID
 }
 
 func (c *container) Info() containers.Container {
+	c.mu.RLock()
+	defer c.mu.RUnlock()
 	return c.c
 }
 
+func (c *container) Extensions() map[string]prototypes.Any {
+	c.mu.RLock()
+	defer c.mu.RUnlock()
+	return c.c.Extensions
+}
+
 func (c *container) Labels(ctx context.Context) (map[string]string, error) {
-	r, err := c.client.ContainerService().Get(ctx, c.ID())
+	c.mu.RLock()
+	defer c.mu.RUnlock()
+	r, err := c.client.ContainerService().Get(ctx, c.c.ID)
 	if err != nil {
 		return nil, err
 	}
-
 	c.c = r
```

Comment:
Seems like a write lock is needed. Although I don't think we need to update container here. :)

Reply:
done